src/transaction/buffer.rs

```diff
@@ -158,7 +159,12 @@ impl Buffer {
             .into_iter()
             .map(|(k, v)| KvPair::new(k, v))
             .collect::<Vec<_>>();
-        res.sort_by_cached_key(|x| x.key().clone());
+
+        if reverse {
```

Comment:
I think the integration tests don't break because of this sorting. I add some debug logs and find the origin result from the TiKV is in asceeding order. We are not doing a reverse scan in server.